(BSR) docs(devtools): clean setup (#5315)
pass-culture/pass-culture-app-native: doc/installation/iOS.md

   1. Connect to the Apple Developer account with Xcode.
   2. Get invited to the Apple Developer group by an Admin.
   3. Download the private key via fastlane:

   ```sh
   bundle exec fastlane ios download_certificates --env testing
   ```

   4. When required (multiple times), use the git ssh URL of the private certificates repository (available on 1password in the "Tech" vault): `pass-culture-app-native-certificates`.
   5. Get the `match repo passphrase` on 1password in the "Tech" vault.

### 🚀 Run the app with yarn

- For Apple External Device

  Connect your device to the computer and run:

  ```sh
  yarn ios:testing --device
  ```

  or use the Xcode interface. Choose a scheme (like testing), then a device and click the `Run` button in the toolbar.

- For Xcode Simulator

  Run:

  ```sh
  yarn ios:testing
  ```

  or use the Xcode interface.

  This will also start the metro server. If not, run in another tab:

  ```sh
  yarn start
  ```

### 😤 Troubleshooting

<<<<<<< HEAD
<details>
  <summary>env: node: No such file or directory
=======
### 😤 Troubleshooting

<br />
<details>
  <summary>env: node: No such file or directory 

> >>>>>>> dfb668535 ((BSR) doc(devtools): add some troubleshootings)

Reviewer: ça c'est un conflit qui n'a pas été correctement résolu

si tu veux, j'ai une présentation que tu pourras me demander à la prochaine journée de pratique

il y a plusieurs conflits à résoudre

si tu ne sais pas faire, on peut s'appeler

## Installation - iOS

**After setting up your [environment][1]**, you can follow the steps below to run the **pass Culture** mobile application on iOS.

### Further installation process

If you encounter errors during this setup, refer to the Troubleshooting section at the end of this document.

<br />

- **install `node`**

  - with NVM to have the right version

  ```sh
  brew install nvm
  nvm install
  nvm use
  ```

  - **AND** with Brew to have XCode build's steps (`Bundle React Native code and images` and `Upload Debug Symbols to Sentry`) working

  ```sh
  brew install node
  ```

  - then we will use node v16 and put it by default:

  ```
  nvm install v16
  nvm use v16
  nvm alias default v16
  ```

> 

Reviewer 2: J'ai pas eu besoin de faire ça je crois 🤔 

J'ai l'impression que c'est une redite d'un cas du troubleshooting (  <summary>env: node: No such file or directory 
  Command PhaseScriptExecution failed with a nonzero exit code</summary>), sauf qu'ici, tu présentes qu'une des 2 solutions

> 

Author: c'était ce qu'il y avait déjà. J'ai pas questionné si ça devait rester comme cela. Il s'avère que j'ai réussi à faire toute l'install en étant en v20 donc pas forcément utile j'imagine

<br />

- **install RVM (Ruby Version Manager) and Ruby**

  To install iOS modules (pods), we use CocoaPods (dependency manager). It is built with Ruby, so make sure it is installed on your computer, otherwise :

  <br />

  - install RVM with :

  ```sh
  curl -sSL https://raw.githubusercontent.com/rvm/rvm/master/binscripts/rvm-installer | bash -s stable
  ```

  <br />

  - install a version of Ruby that's 2.7.5 or higher (version of Ruby is indicated in the [Gemfile](/Gemfile) on the root of the repo) with :

  ```sh
  rvm install "version"
  rvm use "version"
  ```

<br />

> - **install bundler**

Reviewer: cette étape nécessite que l'étape précédente soit faite

je ne sais pas de quelle manière il serait préférable de l'indiquer

  - We use [bundler][2] to install our Ruby gems, install it with :

  ```sh
  gem install bundler
  ```

  <br />

  - Now you can install CocoaPods, that will install the gems, with :

  ```sh
  bundle install
  ```

  <br />

- **install the pods, and create the Pods folder**

  - inside the folder `ios` :

  ```sh
  bundle exec pod install
  ```

  <br />

  - you may be asked to run :

  ```sh
  bundle exec pod repo update

>   ```

Reviewer: suggestion: pour éviter toute confusion, je recommande de changer de répertoire pas avec du texte mais dans la commande directement

~~~suggestion
  ```sh
  pushd ios
  bundle exec pod install
  popd
  ```

  you may be asked to run :

  ```sh
  pushd ios
  bundle exec pod repo update
  popd
  ```
~~~

    <br />

  If `bundle exec pod install` didn't work, check in Xcode -> Settings -> Locations if you have a Command line tools installed, then revalidate the Command line tools by clicking it.

<br />

### 🔥 Firebase setup

You will need to add the `GoogleService-Info.plist` file in the `ios` directory. You can get a copy of the testing configuration one through the password manager, or directly through the Firebase console inside `Project Settings`.

<br />

### 🔨Setup Xcode

In order to launch the app in the Simulator or on your external Apple device, you need to follow these steps:

> - Create an [Apple Developer passCulture account](https://developer.apple.com/)
> 
> - You need to use the 14.2 Xcode version, you can download it [here][3] with an Apple Developer account (follow the 1st step).
> 
> - Install a simulator if not already in Xcode
> 
> - Download the iOS certificates to your computer:

Reviewer: les étapes sont ordonnées

suggestion: utiliser une liste ordonnée

```suggestion
1. Create an [Apple Developer passCulture account](https://developer.apple.com/)
1. You need to use the 14.2 Xcode version, you can download it [here][3] with an Apple Developer account
1. Install a simulator if not already in Xcode
1. Download the iOS certificates to your computer:
```

la liste ordonnée suivante devra etre indentée

1. Connect to the Apple Developer account with Xcode.
2. Get invited to the Apple Developer group by an Admin.
3. Download the private key via fastlane:
   ```sh
   bundle exec fastlane ios download_certificates --env testing
   ```
4. When required (multiple times), use the git ssh URL of the private certificates repository (available on 1password in the "Tech" vault) : `pass-culture-app-native-certificates`.
5. Get the `match repo passphrase` on 1password in the "Tech" vault.

<br />

### 🚀 Run the app with yarn

- For Apple External Device

  Connect your device to the computer and run :

  ```sh
  yarn ios:testing --device
  ```

>   or use the Xcode interface. Choose a scheme (your device) and click the _Run_ button in the toolbar.

Reviewer: nittpick: je mets entre backquotes les mots qui sont techniques ou définit par un autre outil

```suggestion
  or use the Xcode interface. Choose a scheme (your device) and click the `Run` button in the toolbar.
```

Reviewer 2: `Choose a scheme (your device)` est confusant je trouve : le scheme, c'est testing, staging ou production ; le device, c'est à côté dans l'interface, mais c'est pas le scheme. Il vaut mieux demander de choisir le scheme (comme testing) puis de sélectionner le device

</details>

<br />
<details>
  <summary>No binary rubies available for: osx/13.5/arm64/ruby-3.0.0.
Continuing with compilation.</summary>

https://stackoverflow.com/questions/42735805/what-does-no-binary-rubies-available-mean

</details>

<br />
<details>

>   <summary> After the 'bundle exec fastlane ios download_certificates --env testing' command, you gave the SSH git repository link and it's not doing anything.</summary>

Reviewer: je ne suis pas du tout certains, mais il me semble que c'est la commande executée

```suggestion
  <summary> After the `bundle exec fastlane ios download_certificates --env testing` command, you gave the SSH git repository link and it's not doing anything.</summary>
```

>     It might be an issue with your ssh (for exemple if you only cloned the repository through http), try to clone the repository elswhere and try again.

Reviewer: suggestion: juste pour etre certain d'enlever toute confusion possible

```suggestion
    It might be an issue with your ssh (for exemple if you only cloned the repository through http), try to clone the repository elswhere using ssh and try again.
```

</details>

 <br />
<details>

>   <summary>/.rvm/gems/ruby-2.7.5/gems/nap-1.1.0/lib/rest/error.rb:76:in `require': cannot load such file -- openssl (LoadError)

Reviewer: nittpick:

```suggestion
  <summary>`/.rvm/gems/ruby-2.7.5/gems/nap-1.1.0/lib/rest/error.rb:76:in 'require': cannot load such file -- openssl (LoadError)`
```
